### PR TITLE
{lib,mpi}[GCC/11.2.0,GCCcore/11.2.0] OpenMPI v4.1.1, PMIx v4.1.0

### PR DESCRIPTION
**Pull request by Micket**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by Micket** [comment]
This is broken, and I'm not sure how to fix the compilation errors.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/9a33fcaf63d34342e4deb5d18283588a for a full test report.

File: easybuild/easyconfigs/p/PMIx/PMIx-4.1.0-GCCcore-11.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+dependencies = [
+    ('libevent', '2.1.12'),
+    ('zlib', '1.2.11'),
+    ('hwloc', '2.5.0'),
+]
+
+configopts = ' --with-libevent=$EBROOTLIBEVENT --with-zlib=$EBROOTZLIB'
+configopts += ' --with-hwloc=$EBROOTHWLOC'
+configopts += ' --enable-pmix-binaries'
+
```
Comment (branfosj) [inline easybuild/easyconfigs/p/PMIx/PMIx-4.1.0-GCCcore-11.2.0.eb:37]:
```suggestion
configopts += ' --disable-man-pages'

```

Disable the man page generation, as that requires Pandoc:
```
*** Man page setup
checking if want man pages... yes
checking for pandoc... no
configure: WARNING: *** Could not find a suitable pandoc on your system.
configure: WARNING: *** You need pandoc >=1.12 to build OpenPMIx man pages.
configure: WARNING: *** See pandoc.org.
configure: WARNING: *** If no man pages are needed, rerun configure with option '--disable-man-pages'.
configure: WARNING: *** NOTE: If you are building from a tarball downloaded from the OpenPMIx GitHub repository, you do not need Pandoc
configure: error: Cannot continue
```

**Comment by Micket** [comment]
Lets keep an eye on https://github.com/open-mpi/ompi/pull/9212 and see if we should patch this (or maybe there will be a 4.1.2 version soon?)

**Comment by branfosj** [comment]
> Lets keep an eye on [open-mpi/ompi#9212](https://github.com/open-mpi/ompi/pull/9212) and see if we should patch this (or maybe there will be a 4.1.2 version soon?)

I've tested and the build succeeds with that fix.

However, when I try to run I hit https://github.com/open-mpi/ompi/issues/8596 and the workaround in https://github.com/open-mpi/ompi/issues/8596#issuecomment-840454444 solves that issue.

**Comment by rhc54** [comment]
FYI: I have posted a fix for https://github.com/open-mpi/ompi/issues/8596 - see https://github.com/open-mpi/ompi/pull/9213

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-11.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    'OpenMPI-4.1.1_fix-bufferoverflow-in-common_ofi.patch',
+    'OpenMPI-4.0.6_remove-pmix-check-in-pmi-switch.patch'
+]
+checksums = [
+    'e24f7a778bd11a71ad0c14587a7f5b00e68a71aa5623e2157bafee3d44c07cda',  # openmpi-4.1.1.tar.bz2
+    # OpenMPI-4.1.1_fix-bufferoverflow-in-common_ofi.patch
+    'a189d834506f3d7c31eda6aa184598a3631ea24a94bc551d5ed1f053772ca49e',
+    # OpenMPI-4.0.6_remove-pmix-check-in-pmi-switch.patch
+    '8acee6c9b2b4bf12873a39b85a58ca669de78e90d26186e52f221bb4853abc4d',
+]
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-11.2.0.eb:21]:
```suggestion
    'a85b09da354916c2a08c24922268e4f25352daf2787c7c90cd5d24ca4f8d892f',  # 9212.patch
    'cd250609f2c7974774a4e8931873bc795c0769581f91530524df516490de4954',  # 9213.patch
]
```

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-11.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+toolchain = {'name': 'GCC', 'version': '11.2.0'}
+
+source_urls = [
+    'https://patch-diff.githubusercontent.com/raw/open-mpi/ompi/pull/',
+    'https://www.open-mpi.org/software/ompi/v%(version_major_minor)s/downloads',
+]
+sources = [SOURCELOWER_TAR_BZ2]
+patches = [
+    'OpenMPI-4.1.1_fix-bufferoverflow-in-common_ofi.patch',
+    'OpenMPI-4.0.6_remove-pmix-check-in-pmi-switch.patch'
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-11.2.0.eb:16]:
```suggestion
    'OpenMPI-4.0.6_remove-pmix-check-in-pmi-switch.patch',
```

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso
CORE_COUNT=16

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
centos8-eb - Linux centos linux 8.3.2011, x86_64, Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz (skylake), Python 3.6.8
See https://gist.github.com/d3cc04cbb39db38cd021dd0f09e7ab50 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
rocky8-eb - Linux rocky linux 8.4, x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), Python 3.6.8
See https://gist.github.com/6843279924234aa93d40948728473c3c for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
jrlogin11.jureca - Linux centos linux 8.3.2011, x86_64, AMD EPYC 7742 64-Core Processor, Python 3.6.8
See https://gist.github.com/989cacec6d14d0a7ffe13abea31ce510 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
jsfl01.jusuf - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7742 64-Core Processor, Python 3.6.8
See https://gist.github.com/0629695e3a48ac3aeeb3e3a7144f38b4 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
jwvis00.juwels - Linux centos linux 8.4.2105, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 3.6.8
See https://gist.github.com/f619ce91c45dd7153ef9a360d02a3bda for a full test report.

**Comment by verdurin** [comment]
@Micket do you consider this to be ready for review yet?

**Comment by Micket** [comment]
Well, did we wanna do something different with the patches or should i download and attach it here?

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2621.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/035e86629bd7ead08818f3b5ea907d16 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/eda91954eee37177a73eccd3b906bfea for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0211u12a.bear.cluster - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.8.5
See https://gist.github.com/adeb61cef3fbb94d9272ff218e0b53df for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/ab2f2f023b876d7354ced09e82b93904 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (2 easyconfigs in total)
node3502.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/df1520c662ebe5a87fcb3c241c97839b for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0305u15a.bear.cluster - Linux RHEL 8.4, POWER, 8335-GTX (power9le), Python 3.6.8
See https://gist.github.com/78b9e6f5d8bda4e42f13f72407fdfcd8 for a full test report.

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-11.2.0.eb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+toolchain = {'name': 'GCC', 'version': '11.2.0'}
+
+source_urls = ['https://www.open-mpi.org/software/ompi/v%(version_major_minor)s/downloads']
+sources = [SOURCELOWER_TAR_BZ2]
+patches = [
+    'OpenMPI-4.1.1_fix-bufferoverflow-in-common_ofi.patch',
+    'OpenMPI-4.0.6_remove-pmix-check-in-pmi-switch.patch',
+    '9212.patch',
+    '9213.patch',
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-4.1.1-GCC-11.2.0.eb:15]: We should rename these patch files, to start with `OpenMPI-4.1.1`, and have a more descriptive name.

Where they come from should be mentioned in the patch header (a link to the PR), not in the filename.

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — CHANGES_REQUESTED** [review]
@Micket I would prefer adding a reference to the upstream PRs in the patches, since the commit ID in the patch is useless (due to rebasing, I guess)

done in https://github.com/Micket/easybuild-easyconfigs/pull/7

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
As discussed, let's merge this so we can test stuff on top of it.

If we want to make changes (like bump to OpenMPI 4.1.2, perhaps), there's still some time for that (next EasyBuild release is expected end of this month).

**Comment by boegel** [comment]
Going in, thanks @Micket!

**Pull request merged**